### PR TITLE
[draft] Feature - Bypass JWT for internal requests

### PR DESCRIPTION
**Pull request by invertigo**
Bypasses JWT authorization for HTTP calls inside of the MDS namespace.  This is done by pushing the authorization step from the Policy object, down to ServiceRoles and ServiceRoleBindings.  Policy will reject an invalid token (authentication), but the audience and issuer check are done in these new objects, as well as what services may be accessed (currently anything in the namespace).

Note that only traffic inside the namespace bypasses JWT auth.  Traffic from another namespace would still require a valid token to access an MDS service.

CAVEAT:
This optionally requires a ClusterRbacConfig to prevent inter-namespace requests from bypassing JWT.  A cluster-wide resource cannot be reliably configured from a helm chart, because other helm charts for the same cluster would try to create or delete the same resource.  This is what the ClusterRbacConfig looks like (add namespaces as needed).
```
apiVersion: "rbac.istio.io/v1alpha1"
kind: ClusterRbacConfig
metadata:
  name: default
spec:
  mode: 'ON_WITH_INCLUSION'
  inclusion:
    namespaces: ["foo"]
```

### CODE REVIEW

File: helm/mds/templates/auth.yaml [hunk 1]
```diff
@@ -25,6 +25,79 @@ spec:
...
+spec:
+  roleRef:
+    kind: ServiceRole
+    name: mds-external-service
+  subjects:
+  {{- range .Values.jwt.audiences }}
+  # allow presenters of JWT tokens based one or more properties of the token
+  # (audience, claims, etc.)
+  # https://archive.istio.io/v1.3/docs/reference/config/authorization/constraints-and-properties/#supported-properties
+  # Lacuna: add as many constraints here as apply
```
Comment (avatarneil) [inline helm/mds/templates/auth.yaml]: No Lacuna in the codez

**Comment by invertigo** [comment]
I'd been putting off merging this because there wasn't a real need for it.  Now that @jwtodd wants to scrape the /metrics endpoint for internal services from prometheus, we need it.

**Comment by janedotx** [comment]
@invertigo Say are we going to merge this? It seems like it might be nice to have now especially because `mds-policy` and `mds-policy-author` will need to speak to `mds-geography-author`.

**Comment by wesleyfsmith** [comment]
I know this has some conflicts now, but I think it would be useful to get this merged. I have some needs for namespace level pods and it would be awesome to bypass the JWT stuff.

**Pull request merged**